### PR TITLE
Update Data Retention Policy Docs

### PR DESCRIPTION
**Pull request by juliannatetreault**
### What
Update the data retention policy and API documentation to reflect the addition of the site admin-level policy.  

### Why
This update adds clarity to the existing data retention policy and API documentation added in the first phase of the data retention policy work, and reflects the newly added site admin-level data retention policy that was added in the most recent phase of the project. 

### Links
- [JIRA](https://hashicorp.atlassian.net/browse/TF-11540)
- [Related Admin Docs PR](https://github.com/hashicorp/ptfe-releases/pull/1040)

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] (N/A) Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] (N/A) Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] (N/A) New pages have metadata (page name and description) at the top.
- [x] (N/A) New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] (N/A) UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx [hunk 1]
```diff
@@ -230,11 +230,15 @@ Configure [VCS providers](/terraform/cloud-docs/vcs) to use in the organization.
 Data retention policies are exclusive to Terraform Enterprise, and not available in Terraform Cloud. <a href="https://developer.hashicorp.com/terraform/enterprise">Learn more about Terraform Enterprise</a>.
 </EnterpriseAlert>
 
-By default, the **Do not auto-delete data** option is enabled for an organization. This option directs Terraform Enterprise to retain data associated with configuration and state versions, but organization owners can define configurable data retention policies that allow Terraform to _soft delete_ the backing data associated with configuration versions and state versions. Soft deleting refers to marking a data object for garbage collection so that Terraform can delete the object after a set number of days.
+There are three types of data retention policies that an organization owner has the ability to set or override: the admin default policy, do not auto-delete, and auto-delete data.
+
+If the admin default policy is selected, all other selections will be disabled by default until the admin default policy checkbox is unchecked.
+
```
Comment (trujillo-adam) [inline website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx:236]:
```suggestion
An organization owner can set or override the following data retention policies: 

- **Admin default policy** 
- **Do not auto-delete** 
- **Auto-delete data**

Setting the data retention policy to **Admin default policy** disables the other data retention policy settings.

```
Ah, OK. This is a little clearer than how it's documented in [PR 1040](https://github.com/hashicorp/ptfe-releases/pull/1040)

File: website/docs/cloud-docs/workspaces/settings/deletion.mdx [hunk 2]
```diff
@@ -98,4 +98,8 @@ Define configurable data retention policies for workspaces to help reduce object
...
 
-The [organization policy](/terraform/enterprise/users-teams-organizations/organizations#destruction-and-deletion) is the default policy applied to workspaces, but members of individual workspaces can override the policy for their workspaces. The workspace policy always overrides the organization policy.
+The [organization policy](/terraform/enterprise/users-teams-organizations/organizations#destruction-and-deletion) is the default policy applied to workspaces, but members of individual workspaces can override the policy for their workspaces.
+
+The workspace policy always overrides the organization policy, and there are three types of data retention policies that a workspace admin has the ability to set or override: the organization default policy, do not auto-delete, and auto-delete data.
```
Comment (trujillo-adam) [inline website/docs/cloud-docs/workspaces/settings/deletion.mdx:103]:
```suggestion
The workspace policy always overrides the organization policy. An workspace owner can set or override the following data retention policies: 

- **Organization default policy** 
- **Do not auto-delete** 
- **Auto-delete data**
```
If these policies names do not correspond to a UI label, then they do not need to be in boldface.

File: website/docs/cloud-docs/workspaces/settings/deletion.mdx [hunk 3]
```diff
@@ -98,4 +98,8 @@ Define configurable data retention policies for workspaces to help reduce object
...
+
+If the organization default policy is selected, all other selections will be disabled by default until the organization default policy checkbox is unchecked.
```
Comment (trujillo-adam) [inline website/docs/cloud-docs/workspaces/settings/deletion.mdx:105]:
```suggestion
Setting the data retention policy to **Organization default policy** disables the other data retention policy settings.
```

**Review by trujillo-adam — COMMENTED** [review]
Do the data retention policy types have labels in the UI or exist as entities within the system? If not, then I think we should refer to, for example, organization owners being able to "set or override the default policy set by the Terraform Cloud admin". Does that nuance make sense?

File: website/docs/cloud-docs/workspaces/settings/deletion.mdx (hunk 2 above)
Comment (juliannatetreault) [reply website/docs/cloud-docs/workspaces/settings/deletion.mdx:103]: They do correspond to the labels, so I will leave them boldface!

File: website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx [hunk 4]
```diff
@@ -230,11 +230,21 @@ Configure [VCS providers](/terraform/cloud-docs/vcs) to use in the organization.
...
-By default, the **Do not auto-delete data** option is enabled for an organization. This option directs Terraform Enterprise to retain data associated with configuration and state versions, but organization owners can define configurable data retention policies that allow Terraform to _soft delete_ the backing data associated with configuration versions and state versions. Soft deleting refers to marking a data object for garbage collection so that Terraform can delete the object after a set number of days.
+An organization owner can set or override the following data retention policies:
+
+- **Admin default policy**
+- **Do not auto-delete**
+- **Auto-delete data**
+
+Setting the data retention policy to **Admin default policy** disables the other data retention policy settings.
+
+If the admin default policy is selected, all other selections will be disabled by default until the admin default policy checkbox is unchecked.
```
Comment (trujillo-adam) [inline website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx:241]:
```suggestion
```
This repeats the previous paragraph

**Review by trujillo-adam — APPROVED** [review]
One more small thing to address. Approving so you can proceed

File: website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx (hunk 4 above)
Comment (juliannatetreault) [reply website/docs/cloud-docs/users-teams-organizations/organizations/index.mdx:241]: Great catch! I've updated this.

**Pull request merged**